src/main/scala/is/hail/expr/Relational.scala

```diff
@@ -1362,7 +1364,10 @@ case class TableImport(paths: Array[String], typ: TableType, readerOpts: TableRe
         _.map { line =>
           region.clear()
 
-          val split = TextTableReader.splitLine(line, readerOpts.separator, readerOpts.quote)
+          val split = {
+            val sp = TextTableReader.splitLine(line, readerOpts.separator, readerOpts.quote)
+            if (useColIndices.isEmpty) sp else sp.zipWithIndex.filter{case (str, index) => useColIndices.contains(index)}.map{case (str, index)=>str}
```

Comment:
zipWithIndex / filter / map are super nice functional constructs, but they have super bad performance because they allocate each intermediate. Our general strategy is to use them for program control flow on the master, but keep them out of data manipulation routines that get run on the workers.